tests/EppoClientTest.php

```diff
@@ -99,215 +49,96 @@ public static function tearDownAfterClass(): void
-        $this->assertNull($client->getBooleanAssignment('subject-10', self::EXPERIMENT_NAME, $subjectAttributes));
-        $this->assertNull($client->getJSONStringAssignment('subject-10', self::EXPERIMENT_NAME, $subjectAttributes));
-        $this->assertNull($client->getParsedJSONAssignment('subject-10', self::EXPERIMENT_NAME, $subjectAttributes));
+        foreach ($testCases as $testFile => $test) {
+            foreach ($test['subjects'] as $subject) {
+                $result = $this->getTypedAssignment($client, VariationType::from($test['variationType']),
+                    $test['flag'], $subject['subjectKey'], $subject['subjectAttributes'], $test['defaultValue']);
+                $this->assertEquals($subject['assignment'], $result, "$testFile ${test['flag']}");
+            }
+        }
     }
 
-    public function testNoGracefulModeThrows()
+    /**
+     * @throws GuzzleException
+     * @throws \Psr\SimpleCache\InvalidArgumentException
+     * @throws Exception|ClientExceptionInterface
+     */
+    private function getTypedAssignment(EppoClient $client, VariationType $type, string $flag, string $subjectKey, array $subject,
+        array|bool|float|int|string $defaultValue): array|bool|float|int|string|null
     {
-        $pollerMock = $this->getPollerMock();
-        $mockConfigRequester = $this->getExperimentConfigurationRequesterMock(self::MOCK_EXPERIMENT_CONFIG, new Exception('config requester error'));
-        $mockLogger = $this->getMockBuilder(LoggerInterface::class)->getMock();
-
-        $subjectAttributes = [['foo' => 3]];
-        $client = EppoClient::createTestClient($mockConfigRequester, $pollerMock, $mockLogger, false);
-
-        $this->expectException(Exception::class);
-        $client->getAssignment('subject-10', self::EXPERIMENT_NAME, $subjectAttributes);
-        $client->getStringAssignment('subject-10', self::EXPERIMENT_NAME, $subjectAttributes);
-        $client->getNumericAssignment('subject-10', self::EXPERIMENT_NAME, $subjectAttributes);
-        $client->getBooleanAssignment('subject-10', self::EXPERIMENT_NAME, $subjectAttributes);
-        $client->getJSONStringAssignment('subject-10', self::EXPERIMENT_NAME, $subjectAttributes);
-        $client->getParsedJSONAssignment('subject-10', self::EXPERIMENT_NAME, $subjectAttributes);
+        return match ($type) {
+            VariationType::STRING => $client->getStringAssignment($flag, $subjectKey, $subject, $defaultValue),
+            VariationType::BOOLEAN => $client->getBooleanAssignment($flag, $subjectKey, $subject, $defaultValue),
+            VariationType::NUMERIC => $client->getNumericAssignment($flag, $subjectKey, $subject, $defaultValue),
+            VariationType::JSON => $client->getJSONAssignment($flag, $subjectKey, $subject, $defaultValue),
+            VariationType::INTEGER => $client->getIntegerAssignment($flag, $subjectKey, $subject, $defaultValue)
+        };
```

Comment:
consider throwing an error for unknown variation types

Reply:
👍